File: analyzers/its/sources/Net5/Net5/S2612.cs
using System.Security.AccessControl;

namespace Net5
{
    public class S2612
    {
        void Foo(FileSecurity fileSecurity)
        {
            FileSystemAccessRule unsafeAccessRule = new ("Everyone", FileSystemRights.FullControl, AccessControlType.Allow);
            fileSecurity.AddAccessRule(unsafeAccessRule);
            fileSecurity.SetAccessRule(unsafeAccessRule);
        }
    }
}

File: analyzers/its/sources/ManuallyAddedNoncompliantIssues.VB/IntentionalFindings/S2612.vb
' <Your-Product-Name>
' Copyright (c) <Year-From>-<Year-To> <Your-Company-Name>
'
' Please configure this header in your SonarCloud/SonarQube quality profile.
' You can also set it in SonarLint.xml additional file for SonarLint or standalone NuGet analyzer.

Imports System.Security.AccessControl

Public Class S2612
    Public Sub TestCases(fileSecurity As FileSecurity, directorySecurity As DirectorySecurity)
        fileSecurity.SetAccessRule(new FileSystemAccessRule("User", FileSystemRights.ListDirectory, AccessControlType.Allow))

        fileSecurity.SetAccessRule(new FileSystemAccessRule("Everyone", FileSystemRights.Write, AccessControlType.Allow)) ' Noncompliant (S2612) {{Make sure this permission is safe.}}

        directorySecurity.RemoveAccessRule(new FileSystemAccessRule("Everyone", FileSystemRights.Write, AccessControlType.Allow))

        directorySecurity.AddAccessRule(new FileSystemAccessRule("Everyone", FileSystemRights.Write, AccessControlType.Allow)) ' Noncompliant
    End Sub
End Class
